static/templyte/src/app/download-template/download-template.component.ts

```diff
@@ -64,6 +74,43 @@ export class DownloadTemplateComponent implements OnInit {
         this.authService.logout(null);
     }
 
+    storeRating(new_rating) {
+        const component = this;
+        console.log("new rating: "+new_rating);
+        let authorUID:string  =  component.user.uid;
+        component.templateRatingsInfoDatabaseRef.once('value').then(snapshot2 => {
+            const old_rating = snapshot2.child(authorUID).val(); //value of previous rating
+            console.log("old_rating: "+old_rating);
+            let hasRated:boolean = old_rating != null;
+            if (hasRated == true) {
+                component.templateDirectoryInfoDatabaseRef.child('/numberRatings').transaction(function(numberRatings){
+                    return numberRatings - 1;
```

Comment:
What's happening here? I though we agreed the user would be able to replace, but not withdraw ratings.

Reply:
I dont recall that, but if the user can't read the ratings how will we read from the database to get the information to perform the transaction to update it based on its current value and also to later populate mytemplates?